lib/core/unittest/CStringUtilsTest.cc

```diff
@@ -969,4 +969,263 @@ BOOST_AUTO_TEST_CASE(testRoundtripMaxDouble) {
     }
 }
 
+BOOST_AUTO_TEST_CASE(testMemorySizeStringToMB) {
+    bool parsedOk{false};
+    std::size_t memorySizeMB{0};
+    const std::size_t defaultValue{1};
+
+    {
+        // All valid and specifying a whole number of MB
+        std::tie(memorySizeMB, parsedOk) =
+            ml::core::CStringUtils::memorySizeStringToMB("4194304b", defaultValue);
+        BOOST_TEST_REQUIRE(parsedOk);
+        BOOST_TEST_REQUIRE(4, memorySizeMB);
```

Comment:
All the `BOOST_TEST_REQUIRE` macros in this file that have two arguments need to be changed.  Either the comma can be changed to `==`, e.g. `BOOST_TEST_REQUIRE(4 == memorySizeMB)`, or the macro can be changed to `BOOST_REQUIRE_EQUAL`, e.g. `BOOST_REQUIRE_EQUAL(4, memorySizeMB)`.  (I think as it stands the comma is acting as a comma operator, so all these assertions are checking the second argument is not zero, null, false, etc.)